WordPress/twentytwentyone
# TwentyTwentyOne: Restructure button style cascade to improve button color options.

Fixes #858
_Possibly_ fixes: #880

## Summary
Continues with @carolinan work in #879 so im borrowing the summary from there. The cover block sets the links to currentColor so that the text and link color has a high contrast against a chosen palette color. This style was overriding the button block and file block link colors. 

Additionally, this PR attempts to address all Blocks that use a button and use the same border-width style consistently. 
## Test instructions

This PR can be tested by following these steps:

1. Create a post with this [sample content](https://cloudup.com/cXTzNhag6EV)
2. Save and view the front
3. Compare the colors and confirm that all the text is readable.
4. Test hover and focus styles.
5. Turn Dark Mode on or off, test both body background colors.
6. Repeat step 5 with the Block editor.

## Screenshots (UPDATED: 2020-11-25)

Before | After | Before (darkmode) | After (darkmode) 
------------|------------|------------|------------
![image](https://user-images.githubusercontent.com/709581/100164283-8f9ea900-2e85-11eb-99b4-102cd0aad509.png)|![image](https://user-images.githubusercontent.com/709581/100289802-c0471700-2f47-11eb-8fed-e5bc559955bc.png)|![image](https://user-images.githubusercontent.com/709581/100164208-69790900-2e85-11eb-8605-909b605050b2.png)|![image](https://user-images.githubusercontent.com/709581/100289535-22534c80-2f47-11eb-85ac-a3bfc1f5d7d0.png)

## Todo

- [x] Finish solution for nested button colors on a lighter background color.
- [ ] Test and apply this fix to the editor.

## Quality assurance
